src/EditorFeatures/Core/Implementation/InlineRename/IEditorInlineRenameService.cs

```diff
@@ -249,11 +249,13 @@ internal interface IInlineRenameInfoWithFileRename : IInlineRenameInfo
         InlineRenameFileRenameInfo GetFileRenameInfo();
     }
 
+#nullable enable
+
     /// <summary>
     /// Language service that allows a language to participate in the editor's inline rename feature.
     /// </summary>
     internal interface IEditorInlineRenameService : ILanguageService
     {
-        Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int position, CancellationToken cancellationToken);
+        Task<IInlineRenameInfo?> GetRenameInfoAsync(Document document, int position, CancellationToken cancellationToken);
```

Comment:
So IInlineRenameInfo has CanRename as a property; what's the difference of meaning between returning null and returning something with that set to false?

Reply:
No difference. But in order to return something both F# and TS have to define that thing (a subclass), which is just too much unnecessary code.

Reply:
Revisited.